config/environment/ui-test.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+return array(
+
+    'Piwik\Config' => \DI\decorate(function (\Piwik\Config $config) {
+        $config->General['cors_domains'][] = '*';
+        $config->trusted_hosts[] = $config->tests['http_host'];
+        $config->trusted_hosts[] = $config->tests['http_host'] . ':' . $config->tests['port'];
+        return $config;
+    }),
+
+    'observers.global' => \DI\add(array(
+
+        array('Request.dispatch.end', function (&$result) {
+            $request = $_GET + $_POST;
+
+            // Overlay needs the full URLs in order to find the links in the embedded page (otherwise the %
+            // tooltips don't show up)
+            if (!empty($request['method'])
+                && $request['method'] == 'Overlay.getFollowingPages'
```

Comment:
Sorry, I don't really understand re aborting the event etc. If it's too much work should be fine to leave it here I reckon